src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -282,6 +304,14 @@ private static (RoutePattern resolvedRoutePattern, IDictionary<string, string> r
 
             builder.Metadata.Add(action);
 
+            // MVC guarantees that when two of it's endpoints have the same route name they are equivalent.
+            //
+            // However, Endpoint Routing requires Endpoint Names to be unique.
+            if (routeName != null && !suppressLinkGeneration && routeNames.Add(routeName))
```

Comment:
What takes precedence when both exist, route name or endpoint name? (e.g. someone creates a conventional route via MapControllerRoute that has a route name, and then adds EndpointNameMetadata via build). I would expect the endpoint name added via the builder to take precedence.

Can the opposite scenario exist?

Add tests for this scenario.

Reply:
Good point, I'll add some tests for this case.

Reply:
FYI we don't have a way to set endpoint name via attribute right now, builder is the only way to do it.